nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/processors/script/ExecuteScript.java

```diff
@@ -0,0 +1,585 @@
+                            scriptNeedsReload.set(true);
+                        }
+                    } catch (Throwable t) {
+                        modulePath = oldValue;
+                        logger.error(t.getLocalizedMessage(), t);
+                    }
+
+                } else if (SCRIPT_ENGINE.equals(descriptor)) {
+                    // The script engine has changed, so we need to set up a new instance for the selected
+                    // engine name
+                    scriptEngineName = newValue;
+                    setupEngine();
+                }
+            } finally {
+                lock.unlock();
+            }
+        }
+    }
+
+    /**
+     * Creates a classloader to be used by the selected script engine and the provided script file. This
+     * classloader has the ExecuteScript's classloader as a parent (versus the current thread's context
+     * classloader) and also adds the specified module directory to the classpath. This enables scripts
+     * to use other scripts, modules, etc. without having to build them into the ExecuteScript NAR.
+     * If the parameter is null or empty, ExecuteScript's classloader is returned
+     *
+     * @param modulePath The path to a directory containing modules to be used by the script(s)
+     */
+    private ClassLoader createScriptEngineModuleClassLoader(String modulePath) {
+        URLClassLoader newModuleClassLoader = null;
+        if (StringUtils.isEmpty(modulePath)) {
+            return ExecuteScript.class.getClassLoader();
+        }
+        try {
+            newModuleClassLoader =
+                    new URLClassLoader(
+                            new URL[]{new File(modulePath).toURI().toURL()}, ExecuteScript.class.getClassLoader());
+        } catch (MalformedURLException mue) {
+            getLogger().error("Couldn't find modules directory at " + modulePath, mue);
+        }
+        return newModuleClassLoader;
+    }
+
+    /**
+     * Evaluates the given script body (or file) using the current session, context, and flowfile. The script
+     * evaluation expects a FlowFile to be returned, in which case it will route the FlowFile to success. If a script
+     * error occurs, the original FlowFile will be routed to failure. If the script succeeds but does not return a
+     * FlowFile, the original FlowFile will be routed to no-flowfile
+     *
+     * @param context the current process context
+     * @param session the current process session
+     * @throws ProcessException if any error occurs during script evaluation
+     */
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        synchronized (isInitialized) {
+            if (!isInitialized.get()) {
+                createResources();
+            }
+        }
+        ProcessorLog log = getLogger();
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
```

Comment:
since we're giving the script access to the session and since we cannot possible know its intent I say we stop pulling a flowfile for them and let them do it.  In so doing we can remove the input required need as well and let scripts be used as ways to source data as well.  Basically "Here script - we've given you a process session - have fun but you're responsible for accounting for the flow files transfers"